test/floki/html_tree_test.exs

```diff
@@ -256,4 +256,42 @@ defmodule Floki.HTMLTreeTest do
     refute Enum.member?(html_tree, %{html_node3 | type: "marquee"})
     refute Enum.member?(html_tree, 42)
   end
+
+  test "Inspect works with HTMLTree" do
+    html_tree = %HTMLTree{
+      root_nodes_ids: [2, 1],
+      node_ids: [2, 1],
+      nodes: %{
+        1 => %Text{content: "hello", node_id: 1},
+        2 => %Text{content: " world", node_id: 2}
+      }
+    }
+
+    assert inspect(html_tree) ==
+             ~s|%HTMLTree{hello world}|
+  end
+
+  test "Inspect works with HTMLTree with nested nodes" do
+    html_tree = %HTMLTree{
+      root_nodes_ids: [1],
+      node_ids: [6, 5, 4, 3, 2, 1],
+      nodes: %{
+        1 => %HTMLNode{type: "html", children_nodes_ids: [6, 3, 2], node_id: 1},
+        2 => %Comment{content: "start of the stack", node_id: 2, parent_node_id: 1},
+        3 => %HTMLNode{
+          type: "a",
+          attributes: [{"class", "link"}],
+          parent_node_id: 1,
+          children_nodes_ids: [4],
+          node_id: 3
+        },
+        4 => %HTMLNode{type: "b", parent_node_id: 3, children_nodes_ids: [5], node_id: 4},
+        5 => %Text{content: "click me", parent_node_id: 4, node_id: 5},
+        6 => %HTMLNode{type: "span", parent_node_id: 1, node_id: 6}
+      }
+    }
+
+    assert inspect(html_tree) ==
+             ~s|%HTMLTree{<html>start of the stack<a class="link"><b>click me</b></a><span></span></html>}|
```

Comment:
We forgot to format the comment. The suggestions I made is changing that. But the output here will be different.

PS: try to add a `|> dbg()` to line 292 and see the beauty of the output :D